### PR TITLE
feat(task): Task description in the form of comments

### LINKED ISSUE #22786: "deno task" should support JSDoc as task description
It would be tremendously helpful if we could support this:
```jsonc
// deno.jsonc
{
  "tasks": {
    // Run a dev server
    "dev": "deno run ...",

    // Generate a new subcommand
    "cmd:generate": "deno run ...",
    
    // Some other comment, Lorem ipsum dolor sit amet, consectetur adipiscing 
    // elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. 
    // Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi 
    // ut aliquip ex ea commodo consequat.
    "cmd:update": "deno run ...",
  }
}
```

Currently `deno task` would output:
```shell
Available tasks:
- dev
    deno run ...
- cmd:generate
    deno run ...
- cmd:update
    deno run ...
```

But it could output:
```
Available tasks:
- dev
    // Run a dev server
    deno run ...

- cmd:generate
    // Generate a new subcommand
    deno run ...

- cmd:update
    // Some other comment, Lorem ipsum dolor sit amet, consectetur adipiscing 
    // elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. 
    // Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi 
    // ut aliquip ex ea commodo consequat.
    deno run ...
```

This would make discoverability in new projects a lot better and make coming to an older project a breeze.

### PR DESCRIPTION
**Pull request by nathanwhit**
Closes #22786.

TLDR;
```jsonc
{
  "tasks": {
    // Some comment
    //
    // describing what the task does
    "dev": "deno run -A --watch main.ts"
  }
}
```
```bash
deno task
```
Screenshot 2024-03-27 at 1 43 49 PM

### CODE REVIEW

**Review by bartlomieju — APPROVED** [review]
LGTM, fun!

File: cli/args/mod.rs [hunk 1]
```diff
@@ -796,11 +796,18 @@ impl CliOptions {
       } else {
         None
       };
+    let parse_options = deno_config::ParseOptions {
+      include_task_comments: matches!(
+        flags.subcommand,
+        DenoSubcommand::Task(..)
```
Comment (dsherret) [inline cli/args/mod.rs:802]: FWIW, this could be further optimized to `DenoSubcommand::Task(TaskFlags { task: None, .. })`.

Comment (nathanwhit) [reply cli/args/mod.rs:802]: I thought about that, but we also print available tasks when you try to run a task that doesn't exist. I guess we could just not show the comments (or reparse the config?) but seemed more trouble than it's worth
